pkg/inventory/inventory-client.go

```diff
@@ -493,3 +498,28 @@ func (cic *ClusterClient) hasSubResource(groupVersion, resource, subresource str
 	}
 	return false, nil
 }
+
+// getObjStatus returns the list of object status
+// at the beginning of an apply process.
+func getObjStatus(pruneIds, unionIds []object.ObjMetadata) []actuation.ObjectStatus {
```

Comment:
I realized why this is being generated here. It's because we haven't updated InvAddTask to generate the initial status. I think we probably want to do that in a followup PR. The solution is complicated because we don't actually need a Merge method in the first place. The InvAddTask can be injected with the applyIdDs and pruneIDs which were generated from the merge of the local objects and Inventory objects. So all it needs to do is a Replace.

Reply:
Yes, we can move the logic of computing the diff and object status into invAddTask. Then we can use the Replace function instead of the Merge function. 
We can do it in a separate PR.